jetstack/test-infra
# Update dind to support consuming upstream kubernetes releases

This PR rewrites the dind directory to consume upstream kubernetes images instead of grabbing via Bazel.

It allows us to run a multi node kubernetes cluster in a docker container, which can easily be consumed by developers wanting to run e2e tests locally, or Prow alike.

To test:

```
cd dind
cd base
make build
cd ..
make 1.10.2
./dind-up.sh 1.10.2
```

This should run a single container on your local docker daemon. You can then:

```
docker exec -it <container-id> bash
export KUBECONFIG=/var/kubernetes/admin.conf
kubectl get nodes,pods --all-namespaces
```

Depending on your computer and connection speed, you should have a functioning multi node cluster within a few minutes!
